gammapy/datasets/simulate.py

```diff
@@ -70,25 +79,58 @@ def _evaluate_timevar_source(self, dataset, evaluator, time_axis=None):
 
         Returns
         -------
-        npred : `~gammapy.maps.Map`
+        npred : `~gammapy.maps.RegionNDMap`
             Npred map.
         """
-        energy_axis = MapAxis.from_edges(
-            dataset.geoms["geom"].axes["energy"].edges, name="energy_true"
+        energy_true = dataset.edisp.edisp_map.geom.axes["energy_true"]
+        energy_new = energy_true.upsample(10)
+        target = evaluator.model.spatial_model.position
+        edisp = dataset.edisp.get_edisp_kernel(
+            dataset.geoms["geom"].axes["energy"], position=target
         )
+        region_exposure = dataset.exposure.to_region_nd_map(target)
+
+        if not time_axis:
+            tstart = dataset.gti.time_start
+            tstop = dataset.gti.time_stop
+            nbin = int(((tstop - tstart) / t_delta).to(""))
+            time_axis_eval = TimeMapAxis.from_time_bounds(
+                time_min=tstart,
+                time_max=tstop,
+                nbin=nbin,
+            )
+            time_axis = MapAxis.from_bounds(
```

Comment:
I think it would be could to introduce some convenience like `MapAxis.from_time_map_axis()` which handles the conversion.

Reply:
@Atreyee was working on this but she found some issues that made it not very useful. @Atreyee, maybe you can comment on this.

Reply:
I think it is just `MapAxis.from_edges(time_map_axis.edges)` which works only if `time_map_axis` is contiguous. And, it loses all information about the `scale` and the reference time, so I dont know if its really very useful

Reply:
I agree, it's not very useful for users, it might still improve code readability...